src/els_range.erl

```diff
@@ -17,99 +17,99 @@ compare( #{from := FromA, to := ToA}
   From = {Line, Column},
   To = {Line, Column + length(atom_to_list(F))},
   #{ from => From, to => To };
-range({Line, Column}, implicit_fun, {M, F, A}) ->
+range({Line, Column}, implicit_fun, {M, F, A}, _Data) ->
   From = {Line, Column},
   %% Assumes "fun M:F/A"
   Length = 6 + length(atom_to_list(M) ++ atom_to_list(F) ++ integer_to_list(A)),
   To = {Line, Column + Length},
   #{ from => From, to => To };
-range({Line, Column}, implicit_fun, {F, A}) ->
+range({Line, Column}, implicit_fun, {F, A}, _Data) ->
   From = {Line, Column},
   %% Assumes "fun F/A"
   Length = 5 + length(atom_to_list(F) ++ integer_to_list(A)),
   To = {Line, Column + Length},
   #{ from => From, to => To };
-range({Line, Column}, behaviour, Behaviour) ->
+range({Line, Column}, behaviour, Behaviour, _Data) ->
   From = {Line, Column - 1},
   To = {Line, Column + length("behaviour") + length(atom_to_list(Behaviour))},
   #{ from => From, to => To };
-range({Line, Column}, exports, {ToLine, ToColumn}) ->
+range({Line, Column}, exports, {ToLine, ToColumn}, _Data) ->
   From = {Line, Column - 1},
   To = {ToLine, ToColumn - 1},
   #{ from => From, to => To };
-range(Pos, export_entry, {F, A}) ->
+range(Pos, export_entry, {F, A}, _Data) ->
   get_entry_range(Pos, F, A);
-range(Pos, import_entry, {_M, F, A}) ->
+range(Pos, import_entry, {_M, F, A}, _Data) ->
   get_entry_range(Pos, F, A);
-range({Line, Column}, function, {F, _A}) ->
+range({Line, Column}, function, {F, _A}, _Data) ->
   From = {Line, Column},
   To = {Line, Column + length(atom_to_list(F))},
   #{ from => From, to => To };
-range({Line, _Column}, define, _Define) ->
-  From = {Line, 1},
-  To = From,
+range({Line, Column}, define, Define, _Data) ->
+  From = plus({Line, Column}, "define("),
+  To = plus(From, atom_to_list(Define)),
   #{ from => From, to => To };
-range({Line, Column}, include, Include) ->
+range({Line, Column}, include, Include, _Data) ->
   From = {Line, Column - 1},
   To = {Line, Column + length("include") + length(Include) + 5},
   #{ from => From, to => To };
-range({Line, Column}, include_lib, Include) ->
+range({Line, Column}, include_lib, Include, _Data) ->
   From = {Line, Column - 1},
   To = {Line, Column + length("include_lib") + length(Include) + 5},
   #{ from => From, to => To };
-range({Line, Column}, macro, Macro) when is_atom(Macro) ->
+range({Line, Column}, macro, Macro, _Data) when is_atom(Macro) ->
   From = {Line, Column},
   To = {Line, Column + length(atom_to_list(Macro))},
   #{ from => From, to => To };
-range({Line, Column}, module, _) ->
+range({Line, Column}, module, _, _Data) ->
   From = {Line, Column},
   To = From,
   #{ from => From, to => To };
-range({Line, _Column}, parse_transform, _Define) ->
+range({Line, _Column}, parse_transform, _Define, _Data) ->
   From = {Line, 1},
   To = From,
   #{ from => From, to => To };
-range(Pos, record_access, {Record, Field}) ->
+range(Pos, record_access, Record, Field) ->
```

Comment:
Why not to include the fieldname in the id, instead of passing the `Data` everywhere and ignoring it in all places except this one? Actually, this change does not sounds necessary for macros at all. Maybe revert?

Reply:
The field name doesn't really belong in the `id` though. This makes the handling in the rest of the code much cleaner, and we are able to handle less special cases. I think we should do the same for POIs that have `id`s in the format `{F, A}` vs `{M, F, A}`, unifying them to `{M, F, A}`, which we end up doing anyway when adding them to the references table.

Reply:
If you check the changes in this PR we were discarding the value for `Field` except in `els_range:range/3`.

Reply:
Fair enough.